pkgs/tools/misc/bat/default.nix

```diff
@@ -38,10 +31,51 @@ rustPlatform.buildRustPackage rec {
       --prefix PATH : "${stdenv.lib.makeBinPath [ less ]}"
   '';
 
+  passthru = {
+    tests = { inherit (nixosTests) bat; };
+
+    updateScript = writeScript "update.sh" ''
+      #!${stdenv.shell}
+      set -o errexit
+      PATH=${
+        stdenv.lib.makeBinPath [
+          common-updater-scripts
+          git
+          nixfmt
+          nix
+          coreutils
+          gnused
+          gnugrep
+          gawk
+        ]
+      }
+
+      oldVersion="$(nix-instantiate --eval -E "with import ./. {}; lib.getVersion ${pname}" | tr -d '"')"
+      latestTag="$(git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags git@github.com:sharkdp/bat.git '*' | tail --lines=1 | cut --delimiter='/' --fields=3 | sed 's|^v||g')"
+
+      if [ ! "$oldVersion" = "$latestTag" ]; then
+        update-source-version ${pname} "$latestTag" --version-key=version --print-changes
+        nixpkgs="$(git rev-parse --show-toplevel)"
+        default_nix="$nixpkgs/pkgs/tools/misc/bat/default.nix"
+        dummy_cargo="19vhhxfyx3nrngcs6dvwldnk9h4lvs7xjkb31aj1y0pyawz882h9"
+        old_cargo=$(grep '^\s*cargoSha256 = ' pkgs/tools/misc/bat/default.nix | awk -F= '{print $2}' | tr -d '" ;')
+        sed -i "s|$old_cargo|$dummy_cargo|g" "$default_nix"
+        new_cargo=$(nix-build -A bat 2>&1 | grep sha256 | grep got | awk -F: '{print $3}')
+        sed -i "s|$dummy_cargo|$new_cargo|g" "$default_nix"
```

Comment:
[nix-update](https://github.com/Mic92/nix-update/) can also handle this with less code.